File: 1222_queens_that_can_attack_a_king/solution.py
class Solution:
    def queensAttacktheKing(self, queens: List[List[int]], king: List[int]) -> List[List[int]]:
        pos = king.copy()
        queen_dict = dict()
        for queen in queens:
            queen_dict[tuple(queen)] = True
        flag = 0
        attacks = []
        for i in [-1, 0, 1]:
            for j in [-1, 0, 1]:
                if i == 0 and j == 0:
                    continue
                while((0 <= pos[0] <= 7) and (0 <= pos[1] <= 7) and flag == 0):
                    pos[0] += i
                    pos[1] += j
                    if tuple(pos) in queen_dict:
                        attacks.append(list(pos))
                        flag = 1
                flag = 0
                pos = king.copy()
        return attacks
                

File: 1222_queens_that_can_attack_a_king/solution.cpp
    class Solution {
public:
    vector<vector<int>> queensAttacktheKing(vector<vector<int>>& queens, vector<int>& king) {
        vector<vector<int>> queen_mat(8, vector<int>(8, 0));
        for(vector<int> i : queens){
            queen_mat[i[0]][i[1]] = 1;
        }
        vector<int> dir_x = {-1, 0, 1};
        vector<int> dir_y = {-1, 0, 1};
        int flag = 0;
        vector<vector<int>> attacks;
        vector<int> pos = king;
        for(int i : dir_x){
            for(int j : dir_y){
                if(i == 0 && j == 0)
                    continue;
                while(pos[0] >= 0 && pos[0] <= 7 && pos[1]>=0 && pos[1]<=7 && flag == 0){
                    if(queen_mat[pos[0]][pos[1]] == 1){
                        attacks.push_back(pos);
                        flag = 1;
                    }
                    pos[0] += i;
                    pos[1] += j;
                }
                pos = king;
                flag = 0;
            }
        }
        return attacks;
    }
};


